docs/examples/ReinterpretLog.md

```diff
@@ -0,0 +1,194 @@
+# Reinterpret a logging effect
+
+This example shows how to reinterpret a simple, first-order `logging` effect,
+in terms of itself, in order to change the type of the values it logs.
+
+- First, we will define a structured log message type, which is the type our
+  application prefers to log in.
+
+- Next, we will define a logging carrier that prints strings to `stdout`.
+
+- Finally, we will bridge the two with an effect carrier that reinterprets
+  structured log messages as strings.
```

Comment:
I see from the main changes in the README that this adds the example files as first-class documentation. I think this is a nice idea, as the rendering on this website for a nice Markdown or `.lhs` file will be better. It’s not hugely important to me what form the examples take, whether they’re `.hs` or `.lhs` or `.md`, as long as they render nicely in-browser, but it _is_ important that they not be duplicated in-tree, since that’s how things drift out of sync, and having internally inconsistent docs is worse than no docs at all. So as it stands I would be a no on this change. However, I think you can still thread this particular needle with the `markdown-unlit` tool. Furthermore, removing this duplication will make your Nix story shorter and simpler, since you won’t need the `writeDocs` stage, and I think you can drop your `lima` dependency, if not more.

Here I would suggest copying what we do with the compilable `README.md`: use `-pgmL markdown-unlit`, and specify the thus-created .lhs files in the cabal file.

Alternatively, if `.lhs` files render well on this website, that might be a good choice. If the rendered document is comparable to a native Markdown file I would actually probably be in favor of `.lhs`, unless there’s some other concrete reason to choose Markdown.

Reply:
It looks like @patrickt's suggestions would need to be applied to the main README.md/README.lhs document as well.